scons/site_tools/xpcc.py

```diff
@@ -198,11 +198,32 @@ def c_string_literal(env, string):
 		Escapes string and adds quotes.
 	"""
 	# Warning: Order Matters! Replace '\\' first!
-	e = [("\\", "\\\\"), ("\'", "\\\'"), ("\"", "\\\""), ("\t", "\\t"), ("\n", "\\n")]
+	e = [("\\", "\\\\"), ("\'", "\\\'"), ("\"", "\\\""), ("\t", "\\t"), ("\n", "\\n"), ("\f", ""), ("\r", "")]
 	for r in e:
 		string = string.replace(r[0], r[1])
 	return "\"" + string + "\""
 
+def detect_system(env):
+	"""
+		Provide an unified method to determine running platform.
+	"""
+	# will return lowcase names so it can be used building paths
+
+	system_name = platform.system()
+	standards = {'Darwin': 'darwin', 'Linux': 'linux',
+						'Windows': 'windows' }
+
+	if system_name in standards:
+		return standards[platform.system()]
+
+	if system_name.startswith('MSYS') or system_name.startswith('CYGWIN') or system_name.startswith('MINGW'):
+		return 'windows'
+
+	# if this fails, don't bother continuing, it can only escaladate
+	env.Error("Could not detect the running platform, " + system_name + " unsupported. " \
+		"Currently  supported by xpcc : Linux, Windows, Darwin, MSYS, Cygwin")
+	env.Exit(1)
+
```

Comment:
shouldn't the be more along the lines of `Currently  supported by xpcc : Linux, Windows (MSYS, Cygwin), Darwin`